### PR TITLE
fix(nuxt): add `exclude` paths to nitro `tsconfig.server.json`

### PR DESCRIPTION
**Pull request by huseeiin**
Fixed `Cannot write file '/Users/{USERNAME}/nuxt-app/dist/_nuxt/A.vue.62ba23ef.js' because it would overwrite input file.`

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxt/src/core/nitro.ts [hunk 1]
```diff
@@ -112,10 +112,14 @@ export async function initNitro (nuxt: Nuxt & { _nitro?: Nitro }) {
       strict: true,
       generateTsConfig: true,
       tsconfigPath: 'tsconfig.server.json',
-      tsConfig: {
-        include: [
-          join(nuxt.options.buildDir, 'types/nitro-nuxt.d.ts')
-        ]
+      typescript: {
```
Comment (danielroe) [inline packages/nuxt/src/core/nitro.ts:115]: I think you may accidentally have introduced an extra layer of nesting.

**Review by danielroe — CHANGES_REQUESTED** [review]

File: packages/nuxt/src/core/nitro.ts [hunk 2]
```diff
@@ -112,10 +112,14 @@ export async function initNitro (nuxt: Nuxt & { _nitro?: Nitro }) {
...
+        strict: true,
+        generateTsConfig: true,
+        tsconfigPath: "tsconfig.server.json",
+        tsConfig: {
+          include: [join(nuxt.options.buildDir, "types/nitro-nuxt.d.ts")],
+          exclude: ["../dist", "../.output"]
```
Comment (danielroe) [inline packages/nuxt/src/core/nitro.ts:121]: We also will need to exclude node_modules if explicitly mentioning exclude paths

Comment (xxxhussein) [reply packages/nuxt/src/core/nitro.ts:121]: I'm unfamiliar with the repo. I just saw this bug while developing a nuxt app. You probably can make those exclude paths better :)

**Review by danielroe — APPROVED** [review]

File: packages/nuxt/src/core/nitro.ts [hunk 3]
```diff
@@ -115,6 +115,11 @@ export async function initNitro (nuxt: Nuxt & { _nitro?: Nitro }) {
       tsConfig: {
         include: [
           join(nuxt.options.buildDir, 'types/nitro-nuxt.d.ts')
+        ],
+        exclude: [
+          ...nuxt.options.modulesDir.map(m => relativeWithDot(nuxt.options.buildDir, m)),
```
Comment (pi0) [inline packages/nuxt/src/core/nitro.ts:120]: Note: Consider `modulesDir` can technically be normal directories with parent/sibiling `node_modules` dir) as well. We might have filtered out the ones not ending with `node_modules` here for greater safety.

Comment (danielroe) [reply packages/nuxt/src/core/nitro.ts:120]: That's a nice idea. I've just copied across our current default in the nuxt tsconfig so if we filter it further we should update them in parallel.

**Pull request merged**